packages/contracts/test/EmailSignerFactory.t.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "forge-std/Test.sol";
+import "../src/EmailSignerFactory.sol";
+
+contract EmailSignerFactoryTest is Test {
+    EmailSignerFactory factory;
+    EmailSigner implementation;
+    bytes32 constant TEST_ACCOUNT_SALT =
+        keccak256(abi.encodePacked("test@example.com", "123secret"));
+    address deployer = makeAddr("deployer");
+    address dkim = makeAddr("dkim");
+    address verifier = makeAddr("verifier");
+
+    function setUp() public {
+        vm.startPrank(deployer);
+
+        // Deploy implementation
+        implementation = new EmailSigner();
+
+        // Deploy factory
+        factory = new EmailSignerFactory(
+            address(implementation),
+            address(dkim),
+            address(verifier)
+        );
+
+        vm.stopPrank();
+    }
+
+    function testConstructorRevertInvalidImplementation() public {
+        vm.expectRevert("Invalid implementation");
+        new EmailSignerFactory(address(0), address(dkim), address(verifier));
+    }
+
+    function testConstructorRevertInvalidDkimRegistry() public {
+        vm.expectRevert("Invalid DKIM registry");
+        new EmailSignerFactory(
+            address(implementation),
+            address(0),
+            address(verifier)
+        );
+    }
+
+    function testConstructorRevertInvalidVerifier() public {
+        vm.expectRevert("Invalid verifier");
+        new EmailSignerFactory(
+            address(implementation),
+            address(dkim),
+            address(0)
+        );
+    }
+
+    function testPredictAddress() public {
+        address predicted = factory.predictAddress(TEST_ACCOUNT_SALT);
+        assertTrue(predicted != address(0));
```

Comment:
This assertion isn’t really doing anything, maybe the test below is sufficient? Alternatively, we could test it against an actual predicted address using create2